--- a/configs/samples/logger.conf.sample
+++ b/configs/samples/logger.conf.sample
@@ -131,9 +131,9 @@
 ;
 ;debug => debug
 ;security => security
-console => notice,warning,error
+;console => notice,warning,error
 ;console => notice,warning,error,debug
-messages => notice,warning,error
+;messages => notice,warning,error
 ;full => notice,warning,error,debug,verbose,dtmf,fax
 
 ;syslog keyword : This special keyword logs to syslog facility
