--- a/autoconf/ast_ext_tool_check.m4
+++ b/autoconf/ast_ext_tool_check.m4
@@ -7,15 +7,18 @@ AC_DEFUN([AST_EXT_TOOL_CHECK],
 [
 	if test "x${PBX_$1}" != "x1" -a "${USE_$1}" != "no"; then
 		PBX_$1=0
-		AC_PATH_TOOL(CONFIG_$1, $2, No, [${$1_DIR}/bin:$PATH])
+		if test "x${$1_DIR}" != "x"; then
+			AC_PATH_TOOL(CONFIG_$1, $2, No, [${$1_DIR}/bin:$PATH])
+		else
+			AC_PATH_TOOL(CONFIG_$1, $2, No, [$PATH])
+		fi
 		if test ! "x${CONFIG_$1}" = xNo; then
 			if test x"$3" = x ; then A=--cflags ; else A="$3" ; fi
 			$1_INCLUDE=$(${CONFIG_$1} $A)
-			$1_INCLUDE=$(echo ${$1_INCLUDE} | $SED -e "s|-I|-I${$1_DIR}|g")
+			$1_INCLUDE=$(echo ${$1_INCLUDE} | $SED -e "s|-std=c99||g")
 
 			if test x"$4" = x ; then A=--libs ; else A="$4" ; fi
 			$1_LIB=$(${CONFIG_$1} $A)
-			$1_LIB=$(echo ${$1_LIB} | $SED -e "s|-L|-L${$1_DIR}|g")
 
 			if test x"$5" != x ; then
 				saved_cppflags="${CPPFLAGS}"
